### PR TITLE
feat: try to increase file descriptor soft limit

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

File: ckb-bin/src/subcommand/db_repair.rs [hunk 1]
```diff
@@ -1,7 +1,13 @@
+use crate::helper::raise_fd_limit;
 use ckb_app_config::{ExitCode, RepairArgs};
 use ckb_db::RocksDB;
+use ckb_logger::info;
 
 pub fn db_repair(args: RepairArgs) -> Result<(), ExitCode> {
+    if let Some(limit) = raise_fd_limit() {
```
Comment (quake) [inline ckb-bin/src/subcommand/db_repair.rs:7]: suggest to move `raise_fd_limit` code to `run_app`:
https://github.com/nervosnetwork/ckb/blob/fc94d8c8468b676000a33fa8dc415ac1761a86d5/ckb-bin/src/lib.rs#L48

Comment (yangby-cryptape) [reply ckb-bin/src/subcommand/db_repair.rs:7]: Maybe, the sub-commands without using `RocksDB` don't have to raise `fd_limit`.

Like apps in Android, no one likes those apps which always request unnecessary permissions.

**Review by yangby-cryptape — DISMISSED** [review]
IMHO, I think the `if let { info!(..); }` could be moved into the helper function `raise_fd_limit(..)`, since it's a helper function in `ckb-bin`, not in `ckb-utils`. 

Then the error handle of `fdlimit::raise_fd_limit(..)` could be only one..

File: ckb-bin/src/subcommand/db_repair.rs (hunk 1 above)
Comment (quake) [reply ckb-bin/src/subcommand/db_repair.rs:7]: > Maybe, the sub-commands without using RocksDB don't have to raise fd_limit.
Yes, that's the reason of suggestion to move this code to line 48: before line 48, the subcommands are unrelated to rocksdb.

Comment (zhangsoledad) [reply ckb-bin/src/subcommand/db_repair.rs:7]: > > Maybe, the sub-commands without using RocksDB don't have to raise fd_limit.
> 
> Yes, that's the reason of suggestion to move this code to line 48: before line 48, the subcommands are unrelated to rocksdb.

I think you misunderstanding what @yangby-cryptape said. if this code moves to `run_app` line 48, all subcommands will require raise fd_limit, which he against.

Comment (quake) [reply ckb-bin/src/subcommand/db_repair.rs:7]: the sub-commands init / list_hashes / etc will return early:
https://github.com/nervosnetwork/ckb/blob/fc94d8c8468b676000a33fa8dc415ac1761a86d5/ckb-bin/src/lib.rs#L30-L41

Comment (yangby-cryptape) [reply ckb-bin/src/subcommand/db_repair.rs:7]: > after line 48, all sub-commands run / migrate / export / etc will open rocksdb.

Nope! :stuck_out_tongue_closed_eyes: For example, the line 60:
https://github.com/nervosnetwork/ckb/blob/fc94d8c8468b676000a33fa8dc415ac1761a86d5/ckb-bin/src/lib.rs#L60

Comment (quake) [reply ckb-bin/src/subcommand/db_repair.rs:7]: We may move this subcommand before raise_fd_limit

**Review by yangby-cryptape — DISMISSED** [review]

**Review by quake — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Review by driftluo — APPROVED** [review]

**Comment by driftluo** [comment]
bors r=quake,driftluo

**Review by yangby-cryptape — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors retry

**Comment by quake** [comment]
bors r=quake,driftluo,yangby-cryptape

**Comment by quake** [comment]
bors r=quake,driftluo,yangby-cryptape

**Comment by quake** [comment]
bors r=quake,driftluo,yangby-cryptape

**Pull request merged**